File: Python/Palidrome/palindrome.py
import string

def palindrome_check( str_input ):
    # Set to lowercase and remove all spaces.
    str_input = str_input.lower().replace( " ", "" )
    # Remove all punctuation
    str_input = str_input.translate(str.maketrans('', '', string.punctuation))

    str_length = len(str_input)
    # check length.
    if str_length < 3:
        return False

    i = 0
    j = str_length - 1
    while i < j:
        #print( i, j )
        if  str_input[i] != str_input[j]:
            return False
        i += 1
        j -= 1

    return True

File: palindrome/java/com/rhayashi/Palindrome.java
package com.rhayashi;

public class Palindrome {
    // Check if string is a Palindrome;
    // ignoring spaces, punctuation, and uppercase. 
    public static boolean check( String str ) {
        
        // Remove spaces, set to lowercase & remove punctuation
        str = str.replace(" ", "").toLowerCase().replaceAll( "[^a-z0-9]", "");

        int strLength = str.length();
        // Check length.
        if ( strLength < 3 ) {
            return false;

        }
        int i = 0,
            j = strLength - 1;
        while ( i < j ) {
            if ( str.charAt(i) != str.charAt(j) )
            {
                return false;
            }
            ++i;
            --j;
        }

        return true;
    }
}